server/src/main/java/org/opensearch/index/compositeindex/datacube/Metric.java

```diff
@@ -23,10 +24,18 @@
 public class Metric implements ToXContent {
     private final String field;
     private final List<MetricStat> metrics;
+    private final List<MetricStat> baseMetrics;
 
     public Metric(String field, List<MetricStat> metrics) {
         this.field = field;
         this.metrics = metrics;
+        this.baseMetrics = new ArrayList<>();
+        for (MetricStat metricStat : metrics) {
+            if (metricStat.isDerivedMetric()) {
+                continue;
+            }
+            baseMetrics.add(metricStat);
```

Comment:
the naming is a bit confusing with baseMetrics, since it is also present in MetricStat

Reply:
what's confusing here ? since it holds all base metrics we add it right

Reply:
I think the naming is confusing, but its a nit. I'm fine w it

Reply:
I got your point that MetricStat and Metric has same names but they do represent the same - so keeping this for now.